src/AppBundle/Controller/ReceiptController.php

```diff
@@ -0,0 +1,183 @@
+        $usersWithInactiveReceipts = $this->getDoctrine()->getRepository('AppBundle:User')->findAllUsersWithInactiveReceipts();
+
+        return $this->render('receipt_admin/show_receipts.html.twig', array(
+            'users_with_active_receipts' => $usersWithActiveReceipts,
+            'users_with_inactive_receipts' => $usersWithInactiveReceipts,
+            'current_user' => $this->getUser(),
+        ));
+    }
+
+    public function showIndividualAction(User $user)
+    {
+        $active_receipts = $this->getDoctrine()->getRepository('AppBundle:Receipt')->findActiveByUser($user);
+        $inactive_receipts = $this->getDoctrine()->getRepository('AppBundle:Receipt')->findInactiveByUser($user);
+
+        return $this->render('receipt_admin/show_individual_receipts.html.twig', array(
+            'user' => $user,
+            'active_receipts' => $active_receipts,
+            'inactive_receipts' => $inactive_receipts,
+        ));
+    }
+
+    public function createAction(Request $request)
+    {
+        $receipt = new Receipt();
+        $receipt->setUser($this->getUser());
+
+        $active_receipts = $this->getDoctrine()->getRepository('AppBundle:Receipt')->findActiveByUser($this->getUser());
+        $inactive_receipts = $this->getDoctrine()->getRepository('AppBundle:Receipt')->findInactiveByUser($this->getUser());
+
+        $form = $this->createForm(ReceiptType::class, $receipt);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $isImageUpload = $request->files->get('receipt', ['picture_path']) !== null;
+            if ($isImageUpload) {
+                $path = $this->get('app.file_uploader')->uploadReceipt($request);
+                $receipt->setPicturePath($path);
+            }
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($receipt);
+            $em->flush();
+
+            $this->get('event_dispatcher')->dispatch(ReceiptCreatedEvent::NAME, new ReceiptCreatedEvent($receipt));
+
+            $this->addFlash('success', 'Utlegget ditt har blitt registrert.');
+
+            return $this->redirectToRoute('receipt_create');
+        }
+
+        return $this->render('receipt/my_receipts.html.twig', array(
+            'form' => $form->createView(),
+            'receipt' => $receipt,
+            'active_receipts' => $active_receipts,
+            'inactive_receipts' => $inactive_receipts,
+        ));
+    }
+
+    public function editAction(Request $request, Receipt $receipt)
+    {
+        $user = $this->getUser();
+        $isTeamLeader = $this->get('app.roles')->userIsGranted($user, Roles::TEAM_LEADER);
+
+        if (!$isTeamLeader && ($user !== $receipt->getUser() || !$receipt->isActive())) {
+            // !$isTeamLeader && !($user == $receipt->getUser() && $receipt->isActive())
+            // is maybe clearer
+            throw new AccessDeniedHttpException();
```

Comment:
>The AccessDeniedHttpException bypasses the Symfony Security component and always results in a 403 response. You should throw AccessDeniedException (without Http) instead, so that the Security Component displays a login form.

https://insight.sensiolabs.com/projects/ec066ae0-5658-4365-af0c-4ab7492d2ad7/analyses/878?status=violations#542746069